Doc/tools/templates/customsourcelink.html

```diff
@@ -1,10 +1,22 @@
 {%- if show_source and has_source and sourcename %}
+  <script>
+    document.addEventListener('DOMContentLoaded', () => {
+        const title = document.querySelector('meta[property="og:title"]').content;
+        const elements = document.querySelectorAll('.improvepage');
+        elements.forEach(element => {
+            const url = new URL(element.href);
+            url.searchParams.set('pagetitle', title);
+            element.href = url.toString();
+        });
+    });
+  </script>
   <div role="note" aria-label="source link">
     <h3>{{ _('This page') }}</h3>
     <ul class="this-page-menu">
       <li><a href="{{ pathto('bugs') }}">{% trans %}Report a bug{% endtrans %}</a></li>
+      <li><a class="improvepage" href="{{ pathto('improvepage') }}?pageurl=https://docs.python.org/3/{{ pagename }}.html&pagesource={{ sourcename|replace('.rst.txt', '.rst') }}">{% trans %}Improve this page{% endtrans %}</a></li>
       <li>
-        <a href="https://github.com/python/cpython/blob/main/Doc/{{ sourcename|replace('.rst.txt', '.rst') }}"
+        <a href="https://github.com/python/cpython/blob/main/Doc/{{ sourcename|replace('.rst.txt', '.rst?plain=1') }}"
```

Comment:
```suggestion
        <a href="https://github.com/python/cpython/blob/main/Doc/{{ pagename }}.rst?plain=1"
```
Simplification suggestion; [reference](https://www.sphinx-doc.org/en/master/development/html_themes/templating.html#pagename)

Reply:
Done.